Components/scripts/MapBase.gd

```diff
@@ -281,8 +284,58 @@ func wall(path):
 		changed_tiles[point + up] = true
 		changed_tiles[point + upup] = true
 		changed_tiles[point + downdown] = true
+	
+	return alreadyExistingWalls
+
+func possibleDoors(path):
+	var possibleDoorsInWalls = []
+	var possibleDoors = []
+	
+	if path.size() > 0:
+		var previousMovingHorizontal
```

Comment:
Maybe set to null to make it clear that it needs to be null for the first iteration of the loop?